google/cloud/bigquery/job/base.py

```diff
@@ -633,7 +649,9 @@ def done(self, retry=DEFAULT_RETRY, timeout=None, reload=True):
             self.reload(retry=retry, timeout=timeout)
         return self.state == _DONE_STATE
 
-    def result(self, retry=DEFAULT_RETRY, timeout=None):
+    def result(
+        self, retry: retries.Retry = DEFAULT_RETRY, timeout: float = None
+    ) -> "_AsyncJob":
```

Comment:
Since some job classes do use this implementation without overriding it, I wonder if this is more evidence that we should make `_AsyncJob` public (just `Job`?) / make a separate `abc` type for jobs?

Reply:
If that's not something end users should know about, I'd prefer keeping it private, but I support the renaming idea - "async" is too often automatically associated with `async.io` nowadays.

Ditto for adding an `ABC`, if we decide for that with other class families, such as job configs.